src/mon/OSDMonitor.cc

```diff
@@ -6813,8 +6813,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       err = -EINVAL;
       goto reply;
     }
-    PGMap& pg_map = mon->pgmon()->pg_map;
-    if (!pg_map.pg_stat.count(pgid)) {
+    if (!osdmap.pg_exists(pgid)) {
```

Comment:
These function differently — pg_exists just check's whether it's logically present based on pool and pg_num, but pg_stat looks at whether it's been included in a pgstat message.
I presume that's okay and won't bust something if it's run while the PG is being created?